neurom/fst/_neuronfunc.py

```diff
@@ -187,22 +190,11 @@ def sholl_frequency(nrn, neurite_type=NeuriteType.all, step_size=10):
         having crossed multiple times.
     '''
     nrns = neuron_population(nrn)
-    neurite_filter = is_type(neurite_type)
-
-    min_soma_edge = float('Inf')
-    max_radii = 0
-    neurites_list = []
-    for neuron in nrns:
-        neurites_list.extend(((neurites, neuron.soma.center)
-                              for neurites in neuron.neurites
-                              if neurite_filter(neurites)))
-
-        min_soma_edge = min(min_soma_edge, neuron.soma.radius)
-        max_radii = max(max_radii, np.max(np.abs(bounding_box(neuron))))
-
-    radii = np.arange(min_soma_edge, max_radii + step_size, step_size)
-    ret = np.zeros_like(radii)
-    for neurites, center in neurites_list:
-        ret += sholl_crossings(neurites, center, radii)
-
-    return ret
+    if bins is None:
+        min_soma_edge = min(neuron.soma.radius for neuron in nrns)
+        max_radii = np.max([np.abs(bounding_box(neuron)) for neuron in nrns])
+        bins = np.linspace(min_soma_edge, max_radii, 10)
```

Comment:
10 is a hardcoded number of bins, maybe we could have another argument: `n_bins`, or if `bins` is an int, it is used as n_bins? Not sure what is the best option, but I would definitely allow for an int option, for which the bounds are computed automatically as done if bins is None.